src/niivue.js

```diff
@@ -1326,9 +1356,204 @@ Niivue.prototype.initText = async function () {
+        this.volumes[imageIndex].img = new Uint16Array(ta);
+        break;
+      case image.DT_RGBA32:
+        this.volumes[imageIndex].img = new Uint8Array(ta);
+        break;
+      default:
+        throw "datatype " + image.hdr.datatypeCode + " not supported";
+    }
+
+    this.refreshLayers(
+      this.volumes[imageIndex],
+      imageIndex,
+      this.volumes.length
+    );
+    this.drawScene();
+  }
+
+  return ret;
+};
+Niivue.prototype.processImage = function (imageIndex, cmd, addLayer = true) {
+  // clone so we can update the voxel offset
+  let image = this.volumes[imageIndex].clone();
+  console.log("command is " + cmd);
+
+  const dims = image.hdr.dims;
+  const nx = dims[1]; //number of columns
+  const ny = dims[2]; //number of rows
+  const nz = dims[3]; //number of slices
+  const nt = Math.max(1, dims[4]); //number of volumes
+  const pixDims = image.hdr.pixDims;
+  const dx = pixDims[1]; //space between columns
+  const dy = pixDims[2]; //space between rows
+  const dz = pixDims[3]; //space between slices
+  const dt = pixDims[4]; //time between volumes
+  let bpv = Math.floor(image.hdr.numBitsPerVoxel / 8);
+
+  //allocate WASM null-terminated command string
+  let cptr = this.walloc(cmd.length + 1);
+  this.linearMemory.record_malloc(cptr, cmd.length + 1);
+  let cmdstr = new Uint8Array(cmd.length + 1);
+  for (let i = 0; i < cmd.length; i++) cmdstr[i] = cmd.charCodeAt(i);
+  let cstr = new Uint8Array(this.wasmMemory.buffer, cptr, cmd.length + 1);
+  cstr.set(cmdstr);
+  console.log(cstr);
+  //allocate WASM image data
+  let nvox = nx * ny * nz * nt;
+  let ptr = this.walloc(nvox * bpv);
+  this.linearMemory.record_malloc(ptr, nvox * bpv);
+  let cimg = new Uint8Array(this.wasmMemory.buffer, ptr, nvox * bpv);
+  cimg.set(new Uint8Array(image.img.buffer));
+
+  let ok = this.processNiftiImage(
+    ptr,
+    image.hdr.datatypeCode,
+    nx,
+    ny,
+    nz,
+    nt,
+    dx,
+    dy,
+    dz,
+    dt,
+    cptr
+  );
+
+  // Unintuitive Solution: renew cimg pointer
+  // https://depth-first.com/articles/2019/10/16/compiling-c-to-webassembly-and-running-it-without-emscripten/
+  cimg = new Uint8Array(this.wasmMemory.buffer, ptr, nvox * bpv);
+
+  if (ok != 0) {
+    console.log(" -> '", cmd, " generated a fatal error: ", ok);
+    return;
+  }
+
+  let processedImage = addLayer ? image : this.volumes[imageIndex];
+
+  let isBinaryImage = cmd.toLowerCase().includes("-dog");
+  // deal with binarized image from dog
+  if (isBinaryImage) {
+    cimg = cimg.map((x) => x * 255);
+    processedImage.colorMap = "red";
+  }
+
+  switch (processedImage.hdr.datatypeCode) {
+    case processedImage.DT_UNSIGNED_CHAR:
+      processedImage.img = new Uint8Array(cimg);
+      break;
+    case processedImage.DT_SIGNED_SHORT:
+      processedImage.img = new Int16Array(cimg);
+      break;
+    case processedImage.DT_FLOAT:
+      processedImage.img = new Float32Array(cimg);
+      break;
+    case processedImage.DT_DOUBLE:
```

Comment:
Yikes! The WASM code does not handle DT_DOUBLE yet. It could, but it would either double the size of WASM (handle both FLOAT32 and FLOAT64) or double the RAM requirements.

Reply:
fixed to throw error for DT_DOUBLE now